### PR TITLE
feat: Create statesync.bash

### PR DESCRIPTION
**Pull request by faddat**
## State Sync

This is a working state sync bootstrap for regen and will be useful for device images going forward. 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [x] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Review by anilCSE — COMMENTED** [review]
Thank you @faddat for the pr. Can we use the peers and RPC details from https://github.com/regen-network/mainnet/tree/main/regen-1 instead?

**Comment by faddat** [comment]
Soon as I'm done w/ a migration :)

**Comment by ryanchristo** [comment]
Thanks again @faddat! Are you able to make the changes or would you prefer one of us to do so?

File: scripts/statesync.bash [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+set -uxe
+
+# set environment variables
+export GOPATH=~/go
+export PATH=$PATH:~/go/bin
+
+
+# MAKE HOME FOLDER AND GET GENESIS
+regen init statesync
```
Comment (ryanchristo) [inline scripts/statesync.bash:18]: It would be nice to provide a moniker here as an argument.

Comment (ryanchristo) [reply scripts/statesync.bash:18]: Also, we need to include the chain id here.

File: scripts/statesync.bash [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+
+# expor state sync vars
+export REGEN_STATESYNC_ENABLE=true
+export REGEN_P2P_MAX_NUM_OUTBOUND_PEERS=200
+export REGEN_STATESYNC_RPC_SERVERS="http://public-rpc.regen.vitwit.com:26657,http://rpc.regen.forbole.com:80,https://regen.stakesystems.io:2053"
+export REGEN_STATESYNC_TRUST_HEIGHT=$BLOCK_HEIGHT
+export REGEN_STATESYNC_TRUST_HASH=$TRUST_HASH
+export REGEN_P2P_SEEDS="aebb8431609cb126a977592446f5de252d8b7fa1@104.236.201.138:26656"
+export REGEN_P2P_PERSISTENT_PEERS="69975e7afdf731a165e40449fcffc75167a084fc@104.131.169.70:26656,d35d652b6cb3bf7d6cb8d4bd7c036ea03e7be2ab@116.203.182.185:26656,ffacd3202ded6945fed12fa4fd715b1874985b8c@3.98.38.91:26656"
```
Comment (ryanchristo) [inline scripts/statesync.bash:42]: I'm receiving failed dials with these peers. Any ideas? @anilCSE

**Review by ryanchristo — COMMENTED** [review]
Excited to get this merged and to add some instructions to our docs. I'm currently not able to get this working though. I'm receiving failed dial and connection errors. Have you tested this @faddat? Maybe I'm missing something with my setup.

File: scripts/statesync.bash (hunk 2 above)
Comment (ryanchristo) [reply scripts/statesync.bash:41]:
```suggestion
export REGEN_P2P_PERSISTENT_PEERS="aebb8431609cb126a977592446f5de252d8b7fa1@regen.rpc.vitwit.com:26656" 
```

**Review by anilCSE — APPROVED** [review]

**Review by ryanchristo — APPROVED** [review]
Thanks @kaustubhkapatral and @anilCSE!

**Pull request merged**